napalm/ios/ios.py

```diff
@@ -534,14 +566,25 @@ def commit_config(self, message="", revert_in=None):
                 msg = "Candidate config could not be applied\n{}".format(output)
                 raise ReplaceConfigException(msg)
             elif "%Please turn config archive on" in output:
-                msg = "napalm-ios replace() requires Cisco 'archive' feature to be enabled."
-                raise ReplaceConfigException(msg)
+                raise CommitConfirmException(ARCHIVE_DISABLED_MESSAGE)
```

Comment:
You need a check for `if revert_in` here. In other words, probably should look something like:

```python
            elif "%Please turn config archive on" in output:
                if revert_in:
                    raise CommitConfirmException(ARCHIVE_DISABLED_MESSAGE)
                else:
                    msg = "napalm-ios replace() requires Cisco 'archive' feature to be enabled."
                    raise ReplaceConfigException(msg)
```

NAPALM needs archive to be enabled regardless of whether you are using commit-confirm.